presto-native-execution/presto_cpp/main/operators/tests/UnsafeRowShuffleTest.cpp

```diff
@@ -974,21 +977,33 @@ TEST_F(UnsafeRowShuffleTest, partitionAndSerializeOperator) {
   });
 
   auto plan = exec::test::PlanBuilder()
-                  .values({data}, true)
+                  .values({data}, false)
                   .addNode(addPartitionAndSerializeNode(4, false))
                   .planNode();
 
   testPartitionAndSerialize(plan, data);
 }
 
+TEST_F(UnsafeRowShuffleTest, partitionAndSerializeWithLargeInput) {
```

Comment:
Can you add test case
1. output buffer size is one byte and expect one row at a time for output
2. output buffer size is extremely large and expect a single output row
3. check the number of output rows limit and combine together

Reply:
Updated test case 1/ and 2/. Can you elaborate what do you mean by the 3rd?

I have added additional assert in line https://github.com/vermapratyush/presto/blob/91b91d6524a85784d41613bb1ffb297903a98809/presto-native-execution/presto_cpp/main/operators/tests/UnsafeRowShuffleTest.cpp#L422 to verify the expected number of output batches are produced.